botocore/paginate.py

```diff
@@ -211,7 +211,9 @@ def _get_next_token(self, parsed):
                 return [None]
         next_tokens = []
         for token in self._output_token:
-            next_tokens.append(token.search(parsed))
+            next_token = token.search(parsed)
+            if next_token:
```

Comment:
JmesPath expression will return None if the next token is not found. Should make consistent with having no token in response by appending None as opposed to just not appending anything to the next token list.

Also should add comment